pkg/bgpv1/manager/reconcilerv2/policies_test.go

```diff
@@ -497,12 +498,6 @@ func Test_MergeRoutePolicies(t *testing.T) {
 
 func SortRouteStatementsByName(statements []*types.RoutePolicyStatement) {
 	slices.SortFunc(statements, func(i, j *types.RoutePolicyStatement) int {
-		if i.Conditions.String() < j.Conditions.String() {
-			return -1
-		} else if i.Conditions.String() > j.Conditions.String() {
-			return 1
-		} else {
-			return 0
-		}
+		return cmp.Compare(i.Conditions.String(), j.Conditions.String())
```

Comment:
Nice!   That's very clean.

Reply:
```suggestion
		return strings.Compare(i.Conditions.String(), j.Conditions.String())
```

FWIW, this could also use `strings.Compare` which is more performant on strings compared to `cmp.Compare`. But given this is test code it doesn't really matter.

Reply:
Thanks @tklauser, that's very cool. Looks like recent change, no more three way comparator in `strings.Compare`. I think we can leave it as is for the test.

Reply:
@tklauser is there any plans to replace places where strings are using cmp.Compare to strings.Compare across cilium repo? Wondering if there is going to be some gains in performance.

Reply:
The change was introduced in Go 1.23, so fairly recent. Agree to leave it as is for the test.

I'm planning to work on a linter check (probably as part of `staticcheck`) to check for places where `strings.Compare` could be used instead of `cmp.Compare`.